sklearn/base.py

```diff
@@ -239,10 +238,10 @@ def set_params(self, **params):
         for key, value in params.items():
             key, delim, sub_key = key.partition("__")
             if key not in valid_params:
+                local_valid_params = sorted(self.get_params(deep=False).keys())
```

Comment:
get_params already sorts so is this aditional sorting necessary ?

Reply:
Good point.